### DISCUSSION TITLE
Need help from GitHub Copilot

### DISCUSSION THREAD

**Discussion by KolibriSource** [community/community#35355] [Copilot Conversations]
Hello,

Can the GitHub Copilot service create scripts powershell as needed on demand ?
Example: Script to uninstall a Windows application in silent mode

Thank you

**Comment by metal-oopa** [comment]
No. GitHub Copilot tries to understand your intent and to generate the best code it can, but the code it suggests may not always work or even make sense.

It cannot generate the whole script, but if you can break down the script into smaller functions, it can do the remaining task of completing the script.

Below is the code it generated for the example you gave. 
> Script to uninstall a Windows application in silent mode


```pwsh
# Set the application name
$AppName = "Application Name"

# Set the application uninstall string
$UninstallString = "C:\Program Files\MyApp\uninstall.exe /S"

# Set the application uninstall registry key
$UninstallKey = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\MyApp"

# uninstall the application
Start-Process -FilePath $UninstallString -Wait

# remove the application uninstall registry key
Remove-Item -Path $UninstallKey -Recurse -Force

# remove the application folder
Remove-Item -Path "C:\Program Files\MyApp" -Recurse -Force

# remove the application data folder
Remove-Item -Path "C:\ProgramData\MyApp" -Recurse -Force

# remove the application user data folder
Remove-Item -Path "C:\Users\Public\MyApp" -Recurse -Force
```

Hope this answer helps. 🙂
